packages/core/src/client.ts

```diff
@@ -112,6 +113,74 @@ export class TonClient {
         }
     }
 
+    async resolveError(functionName: string, _params: any, err: TonClientError): Promise<TonClientError> {
+        if (err.code !== 23 || !(err.data?.suggest_use_helper_for)) {
+            return err;
+        }
+        try {
+            const [modName, funcName] = functionName.split(".");
+            const api = (await this.client.get_api_reference()).api;
+            
+            const allTypesArray = api.modules.reduce((accumulator: any, element: any) => accumulator.concat(element.types), []);
+            const allTypesDict: { [name: string]: any } = {};
+            allTypesArray.forEach((element: any) => allTypesDict[element.name] = element);
+
+            const module = api.modules.find((x: any) => x.name === modName);
+            const func = module.functions.find((x: any) => x.name === funcName);
+            const param = func.params[1];
+
+            // If there is only context param (or AppObject second param), there is nothing to analyze
+            if (!param || param.generic_name == "AppObject") {
+                return err;
+            }
+
+            const paramTypeInfo = allTypesDict[param.ref_name];
+            walkParameters(paramTypeInfo, _params, "");
+
+            function walkParameters(parameter: any, value: any, path: string) {
```

Comment:
Suggest to rename `parameter` to `valueType` because it is a type of the `value` not a parameter.